File: apps/api/app/schemas/cockpit_in_flight_adjustments.py
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel


class CockpitInFlightSummarySchema(BaseModel):
    headline: str
    total_items: int
    open_positions: int
    open_orders: int
    active_recommendations: int
    watch_only: int
    review_required: int
    high_attention: int


class CockpitInFlightItemSchema(BaseModel):
    id: str
    item_type: Literal["paper_position", "paper_order", "paper_recommendation", "unknown"]
    symbol: str
    asset_id: str | None = None
    asset_name: str | None = None
    asset_detail_path: str | None = None
    has_asset_context: bool = False
    status: str
    opened_at: str | None
    created_at: str | None
    current_state_summary: str
    attention_level: Literal["low", "medium", "high", "unknown"]
    adjustment_label: Literal[
        "watch_only",
        "review_required",
        "stale_data",
        "risk_attention",
        "missing_context",
        "monitor_issue",
        "unknown",
    ]
    reason: str
    evidence: list[str]
    missing_data: list[str]
    recommended_review_action: str
    is_actionable: Literal[False]


class CockpitInFlightNoteSchema(BaseModel):
    title: str
    detail: str
    severity: str
    created_at: str | None


class CockpitInFlightAdjustmentsResponseSchema(BaseModel):
    generated_at: str
    mode: Literal["paper"]
    summary: CockpitInFlightSummarySchema
    items: list[CockpitInFlightItemSchema]
    monitor_notes: list[CockpitInFlightNoteSchema]
    risk_notes: list[str]
    limitations: list[str]
    recommended_review_actions: list[str]

File: apps/web/lib/api/cockpitInFlightAdjustments.ts
import { apiRequest } from "./core";

export type CockpitInFlightItemType = "paper_position" | "paper_order" | "paper_recommendation" | "unknown";
export type CockpitInFlightAttentionLevel = "low" | "medium" | "high" | "unknown";
export type CockpitInFlightAdjustmentLabel =
  | "watch_only"
  | "review_required"
  | "stale_data"
  | "risk_attention"
  | "missing_context"
  | "monitor_issue"
  | "unknown";

export interface CockpitInFlightSummary {
  headline: string;
  total_items: number;
  open_positions: number;
  open_orders: number;
  active_recommendations: number;
  watch_only: number;
  review_required: number;
  high_attention: number;
}

export interface CockpitInFlightItem {
  id: string;
  item_type: CockpitInFlightItemType;
  symbol: string;
  asset_id: string | null;
  asset_name: string | null;
  asset_detail_path: string | null;
  has_asset_context: boolean;
  status: string;
  opened_at: string | null;
  created_at: string | null;
  current_state_summary: string;
  attention_level: CockpitInFlightAttentionLevel;
  adjustment_label: CockpitInFlightAdjustmentLabel;
  reason: string;
  evidence: string[];
  missing_data: string[];
  recommended_review_action: string;
  is_actionable: false;
}

export interface CockpitInFlightNote {
  title: string;
  detail: string;
  severity: string;
  created_at: string | null;
}

export interface CockpitInFlightAdjustmentsResponse {
  generated_at: string;
  mode: "paper";
  summary: CockpitInFlightSummary;
  items: CockpitInFlightItem[];
  monitor_notes: CockpitInFlightNote[];
  risk_notes: string[];
  limitations: string[];
  recommended_review_actions: string[];
}

export async function getCockpitInFlightAdjustments(): Promise<CockpitInFlightAdjustmentsResponse> {
  return apiRequest<CockpitInFlightAdjustmentsResponse>("/cockpit/in-flight-adjustments", {
    method: "GET",
  });
}
